plugin/core/edit.py

```diff
@@ -100,16 +82,24 @@ def changes_sorted(self, changes: 'List[dict]') -> 'List[Dict]':
         #   }
         # ]
 
-        # Maps a change to the tuple (range.start.line, range.start.character)
-        def get_start_position(change):
-            r = change.get('range')
-            start = r.get('start')
+        def get_start_position(ic):
+            index = ic[0]
+            change = ic[1]
+            start = change.get('range').get('start')
             line = start.get('line')
             character = start.get('character')
-            return (line, character)  # Return tuple so comparing/sorting tuples in the form of (1, 2)
-
-        # Sort by start position
-        return sorted(changes, key=get_start_position)
+            return (line, character, index)
+
+        # The spec reads:
+        # > However, it is possible that multiple edits have the same start position: multiple
+        # > inserts, or any number of inserts followed by a single remove or replace edit. If
+        # > multiple inserts have the same position, the order in the array defines the order in
+        # > which the inserted strings appear in the resulting text.
+        # So we sort by start position. But if multiple text edits start at the same position,
+        # we use the index in the array as the key.
+        indexed_changes = [(i, c) for i, c in enumerate(changes)]
+        indexed_changes = sorted(indexed_changes, key=get_start_position, reverse=True)
+        return [ic[1] for ic in indexed_changes]
```

Comment:
I don't know about these lines. It works, but seems a bit sloppy. Is there a faster of more elegant way to express this?

Reply:
you can just do `indexed_changes = sorted(enumerate(changes), key=get_start_position, reverse=True)`, no?

You could also write it like this:

```
def second(tup: 'Tuple[any, any]') -> any:
    return tup[1]

...

return list(map(second, sorted(enumerate(changes), key=get_start_position, reverse=True)))

```

Reply:
In fact I don't think you have to convert to a list at all:

```
return map(second, sorted(enumerate(changes), key=get_start_position, reverse=True))
```

Reply:
... But then you'd have to change the return type of `changes_sorted_in_reverse` from `List[dict]` to `Iterable[dict]`